pySDC/core/Controller.py

```diff
@@ -226,3 +230,33 @@ def hooks(self):
             pySDC.Hooks.hooks: hooks
         """
         return self.__hooks
+
+    def setup_convergence_controllers(self, description):
+        convergence_controller_classes = [CheckConvergence]
+        self.convergence_controllers = []
+
+        if self.params.use_adaptivity:
+            convergence_controller_classes.append(Adaptivity)
+
+        if self.params.use_HotRod:
+            convergence_controller_classes.append(HotRod)
+
+        if 'convergence_controllers' in description.keys():
+            for convergence_controller_class in description['convergence_controllers']:
+                convergence_controller_classes.append(convergence_controller_class)
+
```

Comment:
Why the difference here? You have one default one (`CheckConvergence`) and then custom ones. `Adaptivity` and `HotRod` are currently.. neither/both?